Related issues buildout

claygorman opened the pull request:
copilot:all

claygorman committed:
Add issue linking functionality and parent_id to issues

This commit introduces issue linking functionality. A new table and associated model `issue_links` have been added to allow issues to have various types of links to other issues. It also includes indexes for quicker searching and linking.
In addition, a `parent_id` field was added to the issues table and model, allowing issues to have a parent-child relationship. The concept of linked issues and parent/child issues can help to model complex relationships between tasks.

claygorman committed:
Add issue linking and retrieval functionality

The code now allows Issues to be linked with each other and also fetches linked issues as part of the issue information. Changes have been made on the back-end database models to accommodate this functionality, while the GraphQL implementation has been updated to carry the linked issue details alongside the main issue information.

claygorman committed:
Add search parameters to QueryIssueInput

Two new parameters, search and searchOperator, are now added to the QueryIssueInput in type definitions. This will allow filtering issues based on specific search terms and operators.

claygorman committed:
Implement inverse link types for issues

Added a new property, `inverseLinkType`, to outline the possible inverse link types between issues in `issue-links.js`. This change also required updates to `issue.js` to handle both linked issues and linked by issues. Additionally, the related resolvers have been adjusted to accommodate these changes. This improvement enables a more comprehensive display of issue relationships within the system.

claygorman committed:
Remove default empty string for token

The code for handling the absence of an authorization header has been changed. The previous implementation used an empty string as a default value for the token if it was not found in the request headers. Now, the 'token' variable will be set to 'undefined' if the 'authorization' header is not present.

claygorman commented:
@CodiumAI-Agent /review

QodoAI-Agent commented:
## PR Analysis

- 🎯 **Main theme:** Adding issue linking functionality
- 📝 **PR summary:** This PR introduces the ability to link issues together, allowing for complex relationships between tasks. It includes the creation of a new table and model for issue links, as well as the addition of a parent_id field to the issues table. The PR also includes updates to the GraphQL implementation to carry linked issue details, and the addition of search parameters to the QueryIssueInput.
- 📌 **Type of PR:** Enhancement
- 🧪 **Relevant tests added:** No
- ⏱️ **Estimated effort to review [1-5]:** 4, because the PR introduces significant changes to the database schema, GraphQL queries, and issue model. It requires a good understanding of the existing codebase and the new functionality being introduced.
- 🔒 **Security concerns:** No
## PR Feedback

- 💡 **General suggestions:** The PR introduces a significant feature that enhances the functionality of the application. However, it would be beneficial to include tests to ensure the new functionality works as expected and does not introduce regressions. Additionally, it would be helpful to include more comments in the code to explain the purpose and functionality of the new additions, especially for complex logic.


- <details><summary> 🤖 Code feedback:</summary><hr><table><tr><td>relevant file</td><td>backend/src/db/models/issue-links.js</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><strong>Consider adding a default value for the 'linkType' field. This can help prevent potential issues if a link type is not provided when creating a new issue link. [medium]</strong></td></tr><tr><td>relevant line</td><td><a href=https://github.com/openpro-io/openpro/pull/9/files#diff-ed5a66707480f1dddf356476405bbd3fc88092d091afaf7e9be9ce3caae5216aR31>'+      linkType: {'</a></td></tr></table><hr><table><tr><td>relevant file</td><td>backend/src/db/models/issue.js</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><strong>It might be beneficial to add a foreign key constraint to the 'parentId' field to ensure data integrity. This will ensure that the parent issue exists before a child issue can be created. [important]</strong></td></tr><tr><td>relevant line</td><td><a href=https://github.com/openpro-io/openpro/pull/9/files#diff-75a16dff6f82e031acf615c3269dbf44c9562eef350b83d65640154744f23456R70>'+      parentId: {'</a></td></tr></table><hr><table><tr><td>relevant file</td><td>backend/src/resolvers.js</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><strong>Consider handling potential errors when fetching linked issues. This can help provide a more robust and reliable system. [medium]</strong></td></tr><tr><td>relevant line</td><td><a href=https://github.com/openpro-io/openpro/pull/9/files#diff-eeccd34f21b65d35262196010120c09d08840b9ee627dba4b7a365c998af35fdR488>'+        include: ['</a></td></tr></table><hr><table><tr><td>relevant file</td><td>backend/src/index.js</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><strong>It might be better to handle the case where the 'authorization' header is present but the token is empty or invalid. This can help improve security and prevent unauthorized access. [important]</strong></td></tr><tr><td>relevant line</td><td><a href=https://github.com/openpro-io/openpro/pull/9/files#diff-f37983ff118385e01ef4d7280481cb791b484ab9e47571936e7a1f7e46d24b81R117>'+  const token = request.headers.authorization;'</a></td></tr></table><hr></details>


### How to use

 <details> <summary> Instructions</summary>

> Tag me in a comment '@CodiumAI-Agent' and add one of the following commands:  
> **/review**: Request a review of your Pull Request.   
> **/describe**: Update the PR title and description based on the contents of the PR.   
> **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> **/ask \<QUESTION\>**: Ask a question about the PR.   
> **/update_changelog**: Update the changelog based on the PR's contents.   
> **/add_docs**: Generate docstring for new components introduced in the PR.   
> **/generate_labels**: Generate labels for the PR based on the PR's contents.   
> see the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.

>To edit any configuration parameter from the [configuration.toml](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml), add --config_path=new_value.  
>For example: /review --pr_reviewer.extra_instructions="focus on the file: ..."    
>To list the possible configuration parameters, add a **/config** comment.   

</details>

claygorman commented:
@CodiumAI-Agent /describe

claygorman committed:
Add GitHub Actions workflow for PR agent

A new GitHub Actions workflow has been added. This workflow, defined in pr_agent.yml, is triggered on pull requests and issue comments. It runs a job, pr_agent_job, that uses the PR agent from Codium-ai/pr-agent. It allows the job to write to issues, pull requests, and contents.

claygorman committed:
Add name to GitHub workflow configuration

A name has been added to the `.github/workflows/pr_agent.yml` file to improve clarity. This will help in identifying and managing the workflow associated with pull requests and issue comment activities.

claygorman committed:
Update PR agent workflow configuration

This commit updates the settings in the PR agent workflow file. Specifically, it disables the need for tests review by the PR reviewer and limits the number of code suggestions to 1.

claygorman committed:
Remove PR code suggestions limit from PR agent

The value indicating how many code suggestions the pull request agent should provide was removed. This change makes the PR agent more flexible by not restricting the number of code suggestions it can offer.

claygorman commented:
/describe
